torchao/quantization/quant_primitives.py

```diff
@@ -367,19 +372,29 @@ def _dequantize_affine_no_dtype_check(
     block_size: List[int],
     scale: torch.Tensor,
     zero_point: Optional[torch.Tensor],
-    quant_min: int,
-    quant_max: int,
-    zero_point_domain: str = ZeroPointDomain.INT.name,
+    quant_min: Union[int, float],
+    quant_max: Union[int, float],
+    zero_point_domain: Optional[str] = ZeroPointDomain.INT.name,
     output_dtype: torch.dtype = torch.float32,
 ) -> torch.Tensor:
+    """ This function converts AQT tensors to their high precision floating point representation"""
     assert len(block_size) == input.dim(), f"Got input dim:{input.dim()}, block_size: {block_size}"
     shape_for_reduction, reduction_dims = _get_reduction_params(block_size, input.size())
     original_shape = input.shape
     input = input.view(shape_for_reduction)
     shape_after_reduction = shape_for_reduction
     for i in reduction_dims:
         shape_after_reduction[i] = 1
-    scale = scale.view(shape_after_reduction)
+    scale = scale.view(shape_after_reduction)   
+
+    # This case handles dequantization for float8
+    if zero_point_domain is None:
```

Comment:
similar comment for this one, I feel it probably makes more sense to use `zero_point` here

Reply:
although using `zero_point_domain` has one benefit, that is you can merge this implementation into the branch starting from L401, so I feel that makes sense as well, we just need to document it properly